src/core/device.c

```diff
@@ -152,21 +151,94 @@ static int device_coldplug(Unit *u) {
         return 0;
 }
 
+static const struct {
+        DeviceFound flag;
+        const char *name;
+} device_found_map[] = {
+        [DEVICE_NOT_FOUND] = { DEVICE_NOT_FOUND,     "not-found"     },
+                             { DEVICE_FOUND_UDEV,    "found-udev"    },
+                             { DEVICE_FOUND_UDEV_DB, "found-udev-db" },
+                             { DEVICE_FOUND_MOUNT,   "found-mount"   },
+                             { DEVICE_FOUND_SWAP,    "found-swap"    },
+                             {}
+};
+
+static int device_found_to_string_many(DeviceFound flags, char **ret) {
+        _cleanup_free_ char *s = NULL;
+        unsigned i;
+
+        assert(ret);
+
+        if (flags == DEVICE_NOT_FOUND) {
+                s = strdup(device_found_map[DEVICE_NOT_FOUND].name);
+                if (!s)
+                        return -ENOMEM;
+        } else
```

Comment:
Do we need this special case? If `flags == DEVICE_NOT_FOUND` than the code in the `for` loop should handle this just fine.